File: app/src/main/java/ru/vtb/ppcd/example/route/impl/FromTransformToRoute.java
package ru.vtb.ppcd.example.route.impl;

import ru.vtb.ppcd.example.config.ApplicationConfig;
import ru.vtb.ppcd.example.config.RouteId;
import ru.vtb.ppcd.example.processor.impl.BasicMessageProcessorFunction;
import lombok.AllArgsConstructor;
import org.apache.camel.LoggingLevel;
import org.apache.camel.builder.RouteBuilder;
import org.springframework.stereotype.Component;


@Component
@AllArgsConstructor
public class FromTransformToRoute extends RouteBuilder {
    private final ApplicationConfig applicationConfig;

    @Override
    public void configure() {
        var routePath = applicationConfig.getRoutePathWithIdById(getRouteId());
        from(routePath.in())
                .id(getRouteId().name())
                .log(LoggingLevel.INFO, "офсет - [${header.kafka.OFFSET}], тело - [${body}]")
                .bean(BasicMessageProcessorFunction.class)
                .to(routePath.out());
    }

    private RouteId getRouteId() {
        return RouteId.FROM_TRANSFORM_TO_ID;
    }
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/route/impl/FromTransformToRoute.kt
package org.example.ru.vtb.ppcd.example.route.impl

import org.apache.camel.LoggingLevel
import org.apache.camel.builder.RouteBuilder
import org.example.ru.vtb.ppcd.example.config.RouteConfig
import org.example.ru.vtb.ppcd.example.model.RouteId
import org.example.ru.vtb.ppcd.example.processor.impl.BasicMessageProcessor
import org.springframework.stereotype.Component

@Component
class FromTransformToRoute(
    private val routeConfig: RouteConfig,
) : RouteBuilder() {
    private val routeId: RouteId = RouteId.FROM_TRANSFORM_TERMINATE_ID

    override fun configure(): Unit =
        routeConfig.getRoutePathWithIdById(routeId)
            .let { routePath ->
                from(routePath.`in`)
                    .id(routeId.name)
                    .log(
                        LoggingLevel.INFO, "офсет - [\${header.kafka.OFFSET" +
                                "}], тело - [\${body}]"
                    ).bean(BasicMessageProcessor::class.java)
            }
}